src/Core/Framework/Gateway/Context/Command/ContextGatewayCommandCollection.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Gateway\Context\Command;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Struct\Collection;
+
+/**
+ * @extends Collection<AbstractContextGatewayCommand>
+ */
+#[Package('framework')]
+class ContextGatewayCommandCollection extends Collection
+{
+    public function getSingleTokenCommand(): LoginCustomerCommand|RegisterCustomerCommand|null
+    {
+        /** @var LoginCustomerCommand|RegisterCustomerCommand|null $command */
+        $command = $this->getTokenCommands()->first();
+
+        return $command;
+    }
+
+    public function getTokenCommands(): self
+    {
+        return $this->filter(static fn (AbstractContextGatewayCommand $command) => $command instanceof LoginCustomerCommand || $command instanceof RegisterCustomerCommand);
+    }
```

Comment:
```suggestion
    public function getSingleTokenCommand(): LoginCustomerCommand|RegisterCustomerCommand|null
    {
        $command = $this->getTokenCommands()->first();

        return $command;
    }

    /**
     * @return Collection<LoginCustomerCommand|RegisterCustomerCommand>
     */
    public function getTokenCommands(): Collection
    {
        return $this->filter(static fn (AbstractContextGatewayCommand $command) => $command instanceof LoginCustomerCommand || $command instanceof RegisterCustomerCommand);
    }
```

I prefer this over the typehint. Not sure if it will work though

Reply:
@Bird87ZA I tried this and PHPStan doesnt like it 
Not sure why, thats why I prefered the `self` solution